client/devpi/upload.py

```diff
@@ -334,10 +334,25 @@ def __init__(self, hub, rootpath, origrepo):
         self.rootpath = rootpath
         self.origrepo = origrepo
         self.target_distdir = origrepo.join("dist")
-        python = py.path.local.sysfind("python")
-        if not python:
-            raise ValueError("could not find 'python' executable")
-        self.python = str(python)
+
+    @property
+    def python(self):
+        """
+        Find the best Python executable for invoking builds and uploads.
+        Command line option if used has priority, then a virtualenv if one
+        is activated, but otherwise falls back to sys.executable, the Python
+        under which devpi client is running.
+        """
+        python = self.hub.args.python
```

Comment:
I would have written this as:

```
return self.hub.args.python or self._virtualenv_python() or sys.executable
```

Reply:
Yes, you had that in your original diff. I think it's harder to read though, that's why I split it up.